Templates/TraditionalBridge/InterfaceService.tpl

```diff
@@ -125,61 +125,79 @@ namespace <NAMESPACE>
 
   </IF RETURNS_DATA>
 ;//
-;// --------------------------------------------------------------------------------------------------------------------------------
-;// Define temp fields for date parameters
-;//
-<COUNTER_1_RESET><PARAMETER_LOOP><IF COERCE_DATETIME OR COERCE_DATETIME_NULLABLE><COUNTER_1_INCREMENT></IF></PARAMETER_LOOP><IF COUNTER_1>
-            ;;Temp fields for date parameters
-  <COUNTER_1_RESET>
-  <PARAMETER_LOOP><COUNTER_1_INCREMENT>
-    <IF COERCE_DATETIME OR COERCE_DATETIME_NULLABLE>
-            data dateParam<COUNTER_1_VALUE>, d8
-      <IF IN OR INOUT>
-            SynergyConverter.ConvertBack(args.<PARAMETER_NAME>, dateParam<COUNTER_1_VALUE>, "<PARAMETER_DATE_FORMAT>", ^null)
+;//
+;//
+  <IF DATE_PARAMETERS>
+            ;;Define temporary variables to handle date parameters
+    <PARAMETER_LOOP>
+      <IF DATE>
+            data tmp<PARAMETER_NAME>, d8, 0
       </IF>
+    </PARAMETER_LOOP>
 
-    </IF>
-  </PARAMETER_LOOP>
-</IF>
-;// --------------------------------------------------------------------------------------------------------------------------------
+  </IF>
+;//
+;//
+;//
+  <IF DATE_PARAMETERS_IN>
+            ;;Convert inbound DateTime values to YYYYMMDD
+    <PARAMETER_LOOP>
+      <IF DATE AND IN_OR_INOUT>
+            SynergyConverter.ConvertBack(args.<PARAMETER_NAME>,tmp<PARAMETER_NAME>,"<PARAMETER_DATE_FORMAT>",^null)
+      </IF>
+    </PARAMETER_LOOP>
+
+  </IF DATE_PARAMETERS_IN>
+;//
+;//
+;//
             ;;Make the JSON-RPC call the traditional Synergy routine
             data resultTuple = await CallMethod("<METHOD_NAME>"
-    <COUNTER_1_RESET>
-    <PARAMETER_LOOP><COUNTER_1_INCREMENT>
-      <IF COERCE_DATETIME OR COERCE_DATETIME_NULLABLE>
-            &   ,dateParam<COUNTER_1_VALUE>
```

Comment:
I dont think i understand the difference between datParam<COUNTER_1_VALUE>and tmp<PARAMETER_NAME. do we need both? even if we need both can we avoid the counter pattern and just use some of <PARAMETER_NAME> + somename?

Reply:
Sorry bud, you lost me. The only place I see COUNTER_1 mentioned in the submitted code is in the return value processing to ensure that the resultList is only defined and instantiated one time. The code I changed no longer uses counters. The code you site has a red background, I thought that meant it was removed?

Reply:
I could swear it looked different when i reviewed it but you're right thats the removed code